File: 414-ThirdMaximumNumber/414-ThirdMaximumNumber.py
# Last updated: 7/30/2025, 12:24:00 PM
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        n = set(nums)
        if(len(n)<3):
            return max(n)
        n.remove(max(n))
        n.remove(max(n))
        return max(n)
        

        

File: 414-ThirdMaximumNumber/414-ThirdMaximumNumber.java
// Last updated: 7/30/2025, 12:23:55 PM
class Solution {
    public int thirdMax(int[] nums) {
        Set<Integer> value = new HashSet<>();
        for(int n : nums){
            value.add(n);
        }
        if(value.size()<3){
            return Collections.max(value);
        }
        value.remove(Collections.max(value));
        value.remove(Collections.max(value));
        return Collections.max(value);
    }
}